api/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/metal3-io/cluster-api-provider-metal3/api
 
-go 1.19
+go 1.20
```

Comment:
Not sure if I understood the question, but as you can see gomod is atleast failing since this PR uses go 1.20.

Reply:
Yes, because gomod sees you require 1.20, but we run it in 1.19 container.

Unless you're using Go 1.20 features, you do not need to bump this number, and it'll be compatible with 1.19 as well. Since golangci-lint is moved off the go.mod, and it originally had the 1.19/1.20 compatibility issues, do we have a reason to require 1.20?

Reply:
Yes, we are following CAPI in this regard and by bump to go 1.20 I mean bump this specific requirement. CAPI has bumped the requirement in the [gomod](https://github.com/kubernetes-sigs/cluster-api/blob/main/go.mod#L3).